pkg/api/secretgenerator/secretgenerator.go

```diff
@@ -0,0 +1,140 @@
+package secretgenerator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/getlantern/deepcopy"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/util/gzip"
+)
+
+func LoadConfigFromPath(path string) (Config, error) {
+	var config Config
+	cfgBytes, err := gzip.ReadFileMaybeGZIP(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := yaml.Unmarshal(cfgBytes, &config); err != nil {
+		return nil, err
+	}
+	return config.massiveItemGeneration()
+}
+
+type Config []SecretItem
+
+func (c Config) massiveItemGeneration() (Config, error) {
```

Comment:
```suggestion
func (c Config) expandParameters() (Config, error) {
```

"massive item" is kind of unclear IMHO.

Reply:
Also, what about instead implementing this as [json unmarshaler](https://golang.org/pkg/encoding/json/#Unmarshaler)? That has the advantage that it can't be forgotten

Reply:
That sounds intrested. I'll take a look.